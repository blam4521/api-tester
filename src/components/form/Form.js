import React, { Component } from 'react'

class Form extends Component {
  constructor(props) {
    super(props)
    this.state = {
      method: 'GET',
      url: ''
    }
  }

  handleMethodChange = event => {
    this.setState({ method: event.target.value })
  }

  handleUrlChange = event => {
    this.setState({ url: event.target.value })
  }

  handleSubmit = event => {
    alert(JSON.stringify(this.state))
    event.preventDefault()
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <select
          name="method"
          value={this.state.method}
          onChange={this.handleMethodChange}
        >
          <option value="GET">GET</option>
          <option value="POST">POST</option>
          <option value="PUT">PUT</option>
          <option value="DELETE">DELETE</option>
        </select>
        <input
          type="text"
          name="url"
          placeholder="http://example.com/api"
          value={this.state.url}
          onChange={this.handleUrlChange}
        />
        <input type="submit" value="TEST" />
      </form>
    )
  }
}

export default Form
