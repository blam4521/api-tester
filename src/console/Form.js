import React, { Component } from 'react'
import PropTypes from 'prop-types'

class Form extends Component {
  constructor(props) {
    super(props)
    this.state = {
      method: 'GET',
      url: '',
      body: ''
    }
  }

  handleMethodChange = event => {
    this.setState({ method: event.target.value })
  }

  handleUrlChange = event => {
    this.setState({ url: event.target.value })
  }

  handleBodyChange = event => {
    this.setState({ body: event.target.value })
  }

  handleSubmit = event => {
    this.props.submitFormData(this.state)
    event.preventDefault()
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <select
          name="method"
          value={this.state.method}
          onChange={this.handleMethodChange}
        >
          <option value="GET">GET</option>
          <option value="POST">POST</option>
          <option value="PUT">PUT</option>
          <option value="DELETE">DELETE</option>
        </select>
        <input
          type="text"
          name="url"
          placeholder="http://example.com/api"
          value={this.state.url}
          onChange={this.handleUrlChange}
        />
        {(this.state.method === 'POST' || this.state.method === 'PUT') && (
          <textarea
            name="body"
            rows="5"
            cols="30"
            placeholder={'{\n  "id": 1\n}'}
            value={this.state.body}
            onChange={this.handleBodyChange}
          />
        )}
        <input type="submit" value="TEST" />
      </form>
    )
  }
}

Form.propTypes = {
  submitFormData: PropTypes.func.isRequired
}

export default Form
