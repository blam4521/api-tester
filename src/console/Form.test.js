import React from 'react'
import ReactDOM from 'react-dom'
import TestUtils from 'react-dom/test-utils'
import renderer from 'react-test-renderer'
import Form from './Form'

describe('<Form />', () => {
  it('renders without crashing', () => {
    const submitFormData = jest.fn()
    const div = document.createElement('div')
    ReactDOM.render(<Form submitFormData={submitFormData} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  // full list of assertions here
  // https://facebook.github.io/jest/docs/en/expect.html
  it('contains four options', () => {
    const submitFormData = jest.fn()
    const form = TestUtils.renderIntoDocument(
      <Form submitFormData={submitFormData} />
    )
    const options = TestUtils.scryRenderedDOMComponentsWithTag(form, 'option')

    expect(options).toHaveLength(4)
  })

  it('matches snapshot', () => {
    const submitFormData = jest.fn()
    const form = renderer
      .create(<Form submitFormData={submitFormData} />)
      .toJSON()

    expect(form).toMatchSnapshot()
  })
})
