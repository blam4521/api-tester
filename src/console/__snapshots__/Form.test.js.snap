// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <select
    name="method"
    onChange={[Function]}
    value="GET"
  >
    <option
      value="GET"
    >
      GET
    </option>
    <option
      value="POST"
    >
      POST
    </option>
    <option
      value="PUT"
    >
      PUT
    </option>
    <option
      value="DELETE"
    >
      DELETE
    </option>
  </select>
  <input
    name="url"
    onChange={[Function]}
    placeholder="http://example.com/api"
    type="text"
    value=""
  />
  <input
    type="submit"
    value="TEST"
  />
</form>
`;
